libr/core/disasm.c

```diff
@@ -3565,24 +3565,20 @@ static bool ds_print_core_vmode(RDisasmState *ds, int pos) {
 	case R_ANAL_OP_TYPE_UCALL:
 	case R_ANAL_OP_TYPE_UCALL | R_ANAL_OP_TYPE_REG | R_ANAL_OP_TYPE_IND:
 	case R_ANAL_OP_TYPE_UCALL | R_ANAL_OP_TYPE_IND:
-#if 0
-		if (ds->analop.jump == 0 && ds->analop.ptr) {
-			ut8 buf[sizeof(ut64)] = {0};
-			r_io_read_at (core->io, ds->analop.ptr, buf, sizeof (buf));
-			ut32 n32 = r_read_ble32 (buf, 0);
-			// is valid address
-			// ut32 n64 = r_read_ble32 (buf, 0);
-			ds_print_shortcut (ds, n32, pos);
-		} else {
-			// ds_print_shortcut (ds, ds->analop.jump, pos);
-			ds_print_shortcut (ds, ds->analop.ptr, pos);
-		}
-#endif
 		if (ds->asm_hint_call) {
 			if (ds->analop.jump != UT64_MAX) {
 				slen = ds_print_shortcut (ds, ds->analop.jump, pos);
 			} else {
-				slen = ds_print_shortcut (ds, ds->analop.ptr, pos);
+				ut8 buf[sizeof (ut64)] = {0};
+				r_io_read_at (core->io, ds->analop.ptr, buf, sizeof (buf));
+				ut64 call_addr = core->rasm->bits == 64
+					? r_read_ble64 (buf, core->rasm->big_endian)
+					: r_read_ble32 (buf, core->rasm->big_endian);
+				if (r_io_is_valid_offset (core->io, call_addr, R_PERM_X)) {
+					slen = ds_print_shortcut (ds, call_addr, pos);
+				} else {
```

Comment:
just change call_addr = ds->analop.ptr; in case the offset is not valid and call ds_print_shortcut once